jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/udp/bsm/BsmReceiverTest.java

```diff
@@ -0,0 +1,112 @@
+package us.dot.its.jpo.ode.udp.bsm;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import us.dot.its.jpo.ode.kafka.OdeKafkaProperties;
+import us.dot.its.jpo.ode.kafka.producer.KafkaProducerConfig;
+import us.dot.its.jpo.ode.kafka.topics.RawEncodedJsonTopics;
+import us.dot.its.jpo.ode.test.utilities.EmbeddedKafkaHolder;
+import us.dot.its.jpo.ode.test.utilities.TestUDPClient;
+import us.dot.its.jpo.ode.udp.controller.UDPReceiverProperties;
+import us.dot.its.jpo.ode.util.DateTimeUtils;
+
+@RunWith(SpringRunner.class)
+@EnableConfigurationProperties
+@SpringBootTest(
+    classes = {OdeKafkaProperties.class, UDPReceiverProperties.class, KafkaProducerConfig.class},
+    properties = {
+        "ode.receivers.bsm.receiver-port=15352",
+        "ode.kafka.topics.raw-encoded-json.bsm=topic.BsmReceiverTest"
+    }
+)
+@ContextConfiguration(classes = {
+    UDPReceiverProperties.class,
+    RawEncodedJsonTopics.class, KafkaProperties.class
+})
+class BsmReceiverTest {
+
+  @Autowired
+  UDPReceiverProperties udpReceiverProperties;
+
+  @Autowired
+  RawEncodedJsonTopics rawEncodedJsonTopics;
+
+  @Autowired
+  KafkaTemplate<String, String> kafkaTemplate;
+
+  EmbeddedKafkaBroker embeddedKafka = EmbeddedKafkaHolder.getEmbeddedKafka();
+
+  @Test
+  void testRun() throws Exception {
+    // create the needed topic for production
+    try {
+      embeddedKafka.addTopics(new NewTopic(rawEncodedJsonTopics.getBsm(), 1, (short) 1));
+    } catch (Exception e) {
+      // ignore because we only care that the topics exist not that they're unique
```

Comment:
(non-blocking) This comment seems to indicate that duplicate topics can exist; would this ever be a problem for testing?

Reply:
Sounds good!